evalml/automl/automl_search.py

```diff
@@ -556,13 +565,32 @@ def __init__(
             self.problem_type,
             self.objective,
             self.additional_objectives,
+            self.thresholding_objective,
             self.optimize_thresholds,
             self.error_callback,
             self.random_seed,
             self.X_train.ww.schema,
             self.y_train.ww.schema,
         )
-
+        self.threshold_automl_config = self.automl_config
+        if (
+            is_binary(self.problem_type)
+            and self.optimize_thresholds
+            and self.objective.score_needs_proba
+        ):
+            # use the thresholding_objective
+            self.threshold_automl_config = AutoMLConfig(
+                self.data_splitter,
+                self.problem_type,
+                self.thresholding_objective,
+                self.additional_objectives,
+                self.thresholding_objective,
+                self.optimize_thresholds,
+                self.error_callback,
+                self.random_seed,
+                self.X_train.ww.schema,
+                self.y_train.ww.schema,
+            )
```

Comment:
Yep! The only difference is that the spot for `objective`, we use the `thresholding_objective` instead so that we can train and tune on this when necessary

Reply:
Ah, sorry, I missed the difference there.

Reply:
@bchen1116 hang on, I think @chukarsten 's point still stands. The logic you added in `engine_base.py` looks great. And I see you've added the alternate threshold tuning objective as an additional argument to `AutoMLConfig` above this code block, which looks great. So, why can't you delete this block entirely?

Reply:
Good point! I changed some code to remove this. Hopefully it's simpler!